Sources/StreamChatUI/Composer/ComposerVC.swift

```diff
@@ -627,20 +631,41 @@ open class ComposerVC: _ViewController,
     ///   - typingMention: The potential user mention the current user is typing.
     ///   - mentionRange: The position where the current user is typing a mention to it can be replaced with the suggestion.
     open func showMentionSuggestions(for typingMention: String, mentionRange: NSRange) {
-        userSearchController.search(
-            query: queryForMentionSuggestionsSearch(typingMention: typingMention)
-        )
+        guard let channel = channelController?.channel else {
+            return
+        }
+        guard let currentUserId = channelController?.client.currentUserId else {
+            return
+        }
+        
+        var usersCache: [ChatUser] = []
+
+        if mentionAllAppUsers {
+            userSearchController.search(
+                query: queryForMentionSuggestionsSearch(typingMention: typingMention)
+            )
+        } else {
+            usersCache = searchUsers(
+                channel.watchers.map { $0 } + channel.cachedMembers.map { $0 },
+                by: typingMention,
+                excludingId: currentUserId
+            )
+        }
```

Comment:
All this should be an implementation detail of `ChatUserSearchController`. Ideally, the composer doesn't need to know where the data is coming from and how, it should just send a query to the search controller, and get users.